src/analysis/special_functions/trigonometric/angle.lean

```diff
@@ -361,6 +361,38 @@ begin
   exact cos_pi_div_two_sub _
 end
 
+lemma abs_sin_eq_of_two_nsmul_eq {θ ψ : angle} (h : (2 : ℕ) • θ = (2 : ℕ) • ψ) :
+  |sin θ| = |sin ψ| :=
+begin
+  rw two_nsmul_eq_iff at h,
+  rcases h with rfl | rfl,
+  { refl },
+  { rw [sin_add_pi, abs_neg] }
+end
+
+lemma abs_sin_eq_of_two_zsmul_eq {θ ψ : angle} (h : (2 : ℤ) • θ = (2 : ℤ) • ψ) :
+  |sin θ| = |sin ψ| :=
+begin
+  simp_rw [two_zsmul, ←two_nsmul] at h,
+  exact abs_sin_eq_of_two_nsmul_eq h
+end
+
+lemma abs_cos_eq_of_two_nsmul_eq {θ ψ : angle} (h : (2 : ℕ) • θ = (2 : ℕ) • ψ) :
+  |cos θ| = |cos ψ| :=
+begin
+  rw two_nsmul_eq_iff at h,
+  rcases h with rfl | rfl,
+  { refl },
+  { rw [cos_add_pi, abs_neg] }
+end
+
+lemma abs_cos_eq_of_two_zsmul_eq {θ ψ : angle} (h : (2 : ℤ) • θ = (2 : ℤ) • ψ) :
+  |cos θ| = |cos ψ| :=
+begin
+  simp_rw [two_zsmul, ←two_nsmul] at h,
+  exact abs_cos_eq_of_two_nsmul_eq h
+end
```

Comment:
I think you should use the RHS of `two_nsmul_eq_iff` as the condition for the lemmas and then remove the second version.

Reply:
I think an assertion of equality of twice angles - rather than `ψ = θ ∨ ψ = θ + π` - is most convenient to work with. For example, you can show that a sequence of angles are equal mod `π` in `calc` when working with an equality like that rather than an `or`. (I chose `zsmul` rather than `nsmul` as the main form to use in geometry lemmas on the basis that any time subtraction gets involved, `ℤ` is more convenient than `ℕ`. But for the underlying angle lemmas I provided both `nsmul` and `zsmul` versions, not knowing what other contexts they might get used in or whether `ℤ` or `ℕ` would be involved there.)